### PR TITLE
Drink docs

### PR DESCRIPTION
**Pull request by pmikolajczyk41**

### CODE REVIEW

File: versioned_docs/version-5.x/testing/drink.md [hunk 1]
```diff
@@ -1,20 +1,123 @@
...
 ---
 
 <img src="/img/title/testing1.svg" className="titlePic" />
 
-# Drink
+# DRink!
 
-:::note
-TODO
+Apart from the core ink! testing framework, we also have the [DRink!](https://github.com/inkdevhub/drink) library.
```
Comment (SkymanOne) [inline versioned_docs/version-5.x/testing/drink.md:11]:
```suggestion
Apart from the core ink! testing framework, Cardinal Cryptogtaphy also provides the [DRink!](https://github.com/inkdevhub/drink) library.
```

File: versioned_docs/version-5.x/testing/drink.md [hunk 2]
```diff
@@ -1,20 +1,123 @@
...
+It provides an intermediate solution between integration and E2E testing.
```
Comment (SkymanOne) [inline versioned_docs/version-5.x/testing/drink.md:12]:
```suggestion
It offers an intermediate solution between integration and E2E testing.
```

File: versioned_docs/version-5.x/testing/drink.md [hunk 3]
```diff
@@ -1,20 +1,123 @@
...
 
-Should contain:
+Briefly speaking, DRink! maintains full in-memory blockchain state and allows you to directly interact with it.
```
Comment (SkymanOne) [inline versioned_docs/version-5.x/testing/drink.md:14]:
```suggestion
Intuitively, DRink! maintains a full in-memory blockchain state and allows you to interact with it directly.
```

File: versioned_docs/version-5.x/testing/drink.md [hunk 4]
```diff
@@ -1,20 +1,123 @@
...
 
-* What is it?
-* When to use it?
-* How to use it?
-* Code snippets 
-:::
\ No newline at end of file
+## Comparison to the other testing strategies
+
+To better assess when DRINK performs better than other testing methods, it is crucial to gain a deeper understanding of the consequences of its unique design and the trade-offs it entails.
```
Comment (SkymanOne) [inline versioned_docs/version-5.x/testing/drink.md:20]:
```suggestion
To better assess when DRink! performs better than other testing methods, it is crucial to gain a deeper understanding of the consequences of its unique design and the trade-offs it entails.
```

File: versioned_docs/version-5.x/testing/drink.md [hunk 5]
```diff
@@ -1,20 +1,123 @@
...
+
+1. **Speed:** since DRink! doesn't spawn any node or background process, everything happens locally, within the testing thread.
+This means that the execution can be synchronous and significantly faster than the E2E tests.
+Therefore, its performance should be comparable to the unit or integration testing.
+Also, there is no block production or finalization delay, which is a noticeable factor in the E2E tests.
+Thanks to that, we can launch long-running simulations in a reasonable time.
+
+2. **Testing multiple contracts:** since we are working with a full blockchain state, we can perform any interaction with the contracts, which includes working with mutliple contracts at the same time.
+This is of course the same as in the E2E tests, but not possible in neither unit nor integration tests.
```
Comment (SkymanOne) [inline versioned_docs/version-5.x/testing/drink.md:29]:
```suggestion
Of course, this is the same as in the E2E tests, but it is not possible in either the unit or integration tests.
```

File: versioned_docs/version-5.x/testing/drink.md [hunk 6]
```diff
@@ -1,20 +1,123 @@
...
+
+3. **Working with arbitrary runtimes:** similarly to the E2E tests, where we can spawn any node with customized runtime (Polkadot's term for the state transition function), in DRink! tests we can work with any blockchain we want.
```
Comment (SkymanOne) [inline versioned_docs/version-5.x/testing/drink.md:31]:
```suggestion
3. **Working with arbitrary runtimes:** similarly to the E2E tests, where we can spawn any node with customized runtime (Polkadot's term for the state transition function), in DRink! tests we can work with any blockchain runtime we want.
```

File: versioned_docs/version-5.x/testing/drink.md [hunk 7]
```diff
@@ -1,20 +1,123 @@
...
+
+4. **Full control over runtime state:** we hold the state of the blockchain in hand, so we can easily manipulate it in any way we want.
```
Comment (SkymanOne) [inline versioned_docs/version-5.x/testing/drink.md:33]:
```suggestion
4. **Full control over runtime state:** we hold the state of the blockchain and exercise full control over it, so we can easily manipulate it however we want.
```

File: versioned_docs/version-5.x/testing/drink.md [hunk 8]
```diff
@@ -1,20 +1,123 @@
...
+1. **No node layer:** since we don't spawn any node, we don't have access to the node layer, which means that we can't test any node-related functionality, like RPC calls, block production, etc.
+
+2. **Artificial, isolated environment:** this is the main trade-off of DRink!.
+It might give a false sense of security, while in the real environment, the contract could behave differently.
+The discrepancy can be mitigated by a careful and precise simulation and setup of the environment, but it is still a factor to consider.
+
+3. **No typed contract API:** currently, DRink! works with string-encoded arguments and values, which means that we lose the type safety and convenience that was present in the other testing frameworks.
+Fortunately, this is going to change soon, as there is an ongoing effort to integrate it with [ink-wrapper](https://github.com/Cardinal-Cryptography/ink-wrapper) library.
+
+## When to use DRink!?
```
Comment (SkymanOne) [inline versioned_docs/version-5.x/testing/drink.md:53]:
```suggestion
## When to use `DRink!`?
```

File: versioned_docs/version-5.x/testing/drink.md (hunk 4 above)
Comment (pmikolajczyk41) [reply versioned_docs/version-5.x/testing/drink.md]: :white_check_mark:

File: versioned_docs/version-5.x/testing/drink.md (hunk 5 above)
Comment (pmikolajczyk41) [reply versioned_docs/version-5.x/testing/drink.md]: :white_check_mark:

File: versioned_docs/version-5.x/testing/drink.md (hunk 6 above)
Comment (pmikolajczyk41) [reply versioned_docs/version-5.x/testing/drink.md]: :white_check_mark:

File: versioned_docs/version-5.x/testing/drink.md (hunk 8 above)
Comment (pmikolajczyk41) [reply versioned_docs/version-5.x/testing/drink.md]: :white_check_mark:

**Review by cmichi — APPROVED** [review]
Thank you, Piotr!

File: versioned_docs/version-5.x/testing/drink.md [hunk 9]
```diff
@@ -1,20 +1,123 @@
...
+### As an alternative backend to ink!'s E2E testing framework
+
+DRink! is already integrated with the ink! framework and can be used as a drop-in replacement for the standard E2E testing environment.
+Just use corresponding argument in the test macro:
+```rust
+#[ink_e2e::test(backend(runtime_only()))]
+```
+to your test function and you have just switched from E2E testcase to DRink!-based one, that doesn't use any running node in the background!
+
+For a full example check out [ink! repository](https://github.com/paritytech/ink/blob/master/integration-tests/e2e-runtime-only-backend/lib.rs).
```
Comment (cmichi) [inline versioned_docs/version-5.x/testing/drink.md:112]: I'll update this link to point to https://github.com/paritytech/ink-examples once we've updated that repo to 5.0. It'll then contain the content of the ink! repositories `integration-tests` folder.

**Pull request merged**